src/components/TodoItem.tsx

```diff
@@ -0,0 +1,178 @@
+/* eslint-disable jsx-a11y/label-has-associated-control */
+import React, {
+  useContext,
+  useEffect,
+  useRef,
+  useState,
+} from 'react';
+import classNames from 'classnames';
+import { Todo } from '../types/Todo';
+import { TodosContext } from './TodosContext';
+
+type Props = {
+  todo: Todo,
+  deleteTodo: (id: number) => Promise<unknown>,
+  updatedTodo: (newTodo: Todo) => Promise<unknown>,
+};
+
+export const TodoItem: React.FC<Props> = ({
+  todo,
+  deleteTodo,
+  updatedTodo,
+}) => {
+  const [editValue, setEditValue] = useState(todo.title);
+  const [editTodosId, setEditTodosId] = useState(0);
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  const { todos, setTodos, USER_ID } = useContext(TodosContext);
+
+  useEffect(() => {
+    if (editTodosId === todo.id) {
+      inputRef.current?.focus();
+    }
+  }, [editTodosId, todo.id]);
+
+  const updateTitle = (newTitle: string) => {
+    const trimmedTitle = newTitle.trim();
+
+    if (!trimmedTitle) {
+      deleteTodo(todo.id)
+        .then(() => {
+          const newTodos = todos.filter((t) => t.id !== todo.id);
+
+          setTodos(newTodos);
```

Comment:
```suggestion
          setTodos(currentTodos => currentTodos.filter((t) => t.id !== todo.id));
```
it's better to do it in this way if it's possible: https://react.dev/reference/react/useState#updating-state-based-on-the-previous-state

Reply:
not fixed:
![image](https://github.com/mate-academy/react_todo-app-with-api/assets/86923089/3a5f5c00-d9ac-4abb-9e3b-7c4295dbee56)